Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -673,6 +716,17 @@ public void HandleOrderRequest(OrderRequest request)
             request.SetResponse(response, OrderRequestStatus.Processed);
         }
 
+        /// <summary>
+        /// Loads in cache information of existing orders stored in json file
+        /// </summary>
+        protected void RefreshOpenOrdersCache()
+        {
+            if (File.Exists(OpenOrdersFilePath))
+            {
+                _openOrdersCache = JsonConvert.DeserializeObject<List<SerializedOrder>>(File.ReadAllText(OpenOrdersFilePath));
```

Comment:
Interesting!
Currently Lean is expected to be restoring the orders from the brokerage itself, see brokerage setup handler, I'd expect the order retrieve to be valid/complete, isn't the underlying bug here a brokerage implementation which isn't returning the tags set on the orders (like https://github.com/QuantConnect/Lean/issues/3735) ?

Reply:
It seems that not all brokerages allow to keep the order tags. 
Tradier does seem to have that option https://documentation.tradier.com/brokerage-api/trading/place-equity-order
IB has many additional fields where additional info could be stored , I guess OrderMiscOptions could work for the purpose https://interactivebrokers.github.io/tws-api/classIBApi_1_1Order.html#a83f6142af0b975ae26c85573d0daee2e and it's just that in current IB implementation in Lean tags are not stored/restored.

At the same time I found there are other brokerages, example crypto, that miss that field for the order:
CoinBase https://docs.pro.coinbase.com/#place-a-new-order
Binance https://binance-docs.github.io/apidocs/spot/en/#new-order-trade
so I decided we could add the feature that would work for all the brokerage

_Note: current implementation restores tag for limit orders only , I didn't pay attention to the fact there could also be other types that are submitted and not filled when algo restarts, they are LimitIfTouched and StopOrders I guess, if I am not missing any others? This I will fix._